cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,233 @@
+# Radius Azure Workload Identity Support
+
+* **Status**: In Review
+* **Author**: Will Smith (@willdavsmith)
+
+## Overview
+
+A software workload such as a container-based application, service or script needs an identity to authenticate, access, and communicate with services that are distributed across different platforms and/or cloud providers. For workloads running outside of Azure, the workload need a credential (secret or certificate) to access Azure resources. These credentials pose a security threat and must be stored and rotated regularly.  
+
+Radius supports service principal identity to configure the Azure provider to deploy and manage Azure resources. However, this adds an extra task for operators to rotate the credentials for the identity. A workload identity or federated identity helps us avoid this maintenance challenge of manually managing the credentials and eliminates the risk of exposing secrets or having certificates expire. 
+
+The goal of the scenario is to enable infrastructure operators to configure workload identity support for the Azure provider in Radius to deploy and manage Azure resources.
+
+## Terms and definitions
+
+1. **Service principal** is the local representation of an application object in a tenant or directory. It's the identity of the application instance. Service principals define application access and resources the application accesses.
+
+1. **Managed Identities** are a special type of service principal that eliminate the need for users to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens. These can be used only with the scope of Azure ecosystem. There are two types of managed identities:
+    * **System assigned managed identities**: This managed identity is tied to the lifecycle of the Azure resource and are used to access a particular resource.
+    * **User assigned managed identities**: This managed identity is created as a separate resource or entity in Azure and are assigned to multiple Azure resources.
+
+1. **Workload identities** use the same service principal or managed identity to trust tokens from an external identity provider (IdP). Once that trust relationship is created, your external software workload exchanges trusted tokens from the external IdP for access tokens from the Microsoft Identity platform. Your software workload uses that access token to access the protected Azure resources to which the workload has been granted access. This eliminates the burden of managing and rotating the secrets.
```

Comment:
Are there error conditions that we need to simulate to ensure we are handling the identities correctly?

Reply:
Yes, I can discuss more in the test section.